openpype/pipeline/publish/lib.py

```diff
@@ -354,6 +354,61 @@ def publish_plugins_discover(paths=None):
     return result
 
 
+def _get_plugin_settings(host_name, project_settings, plugin, log):
+    """Get plugin settings based on host name and plugin name.
+
+    Args:
+        host_name (str): Name of host.
+        project_settings (dict[str, Any]): Project settings.
+        plugin (pyliblish.Plugin): Plugin where settings are applied.
+        log (logging.Logger): Logger to log messages.
+
+    Returns:
+        dict[str, Any]: Plugin settings {'attribute': 'value'}.
+    """
+
+    # Use project settings from host name category when available
+    try:
+        return (
+            project_settings
+            [host_name]
+            ["publish"]
+            [plugin.__name__]
+        )
+    except KeyError:
+        pass
+
+    # Category determined from path
+    # - usually path is './<category>/plugins/publish/<plugin file>'
+    filepath = os.path.normpath(inspect.getsourcefile(plugin))
+    filepath = os.path.normpath(filepath)
+
+    split_path = filepath.split(os.path.sep)
```

Comment:
This could be `.rsplit(os.path.sep, 5)` to skip splitting even further than five levels, very minor optimization.
I assume this is called quite often?